sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReceiverClient.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * The builder that creates session-related {@link ServiceBusReceiverClient} instances.
+ * Use {@link #acceptSession(String)} to create a {@link ServiceBusReceiverClient} that is tied to a known specific
+ * session id.
+ * Use {@link #acceptNextSession()} to create a {@link ServiceBusReceiverClient} that is tied to an unknown
+ * available session.
+ * Use {@link #getReceiverClient(int)} to create a {@link ServiceBusReceiverClient} that
+ * process events from up to maxConcurrentSessions number of sessions.
+ */
+public class ServiceBusSessionReceiverClient implements AutoCloseable {
+    private final ServiceBusSessionReceiverAsyncClient sessionAsyncClient;
+    private final Duration operationTimeout;
+
+    ServiceBusSessionReceiverClient(ServiceBusSessionReceiverAsyncClient asyncClient, Duration operationTimeout) {
+        this.sessionAsyncClient = Objects.requireNonNull(asyncClient, "'asyncClient' cannot be null.");
+        this.operationTimeout = Objects.requireNonNull(operationTimeout, "'operationTimeout' cannot be null.");
+    }
+
+    /**
+     * Create a link for the next available session and use the link to create a {@link ServiceBusReceiverClient}
+     * to receive messages from that session.
+     * @return A {@link ServiceBusReceiverClient} that is tied to the available session.
```

Comment:
Add documentation about what it throws if there is no session to accept. Same with async one.

Reply:
Added